src/kagglehub/models.py

```diff
@@ -18,5 +23,29 @@ def model_download(handle: str, path: Optional[str] = None):
     return registry.resolver(h, path)
 
 
-def model_upload():
-    raise NotImplementedError()
+def model_upload(
+    handle: str, local_model_dir: str, license_name: str, model_type: str, version_notes: Optional[str] = None
```

Comment:
Re-posting in case you missed it: https://github.com/Kaggle/kagglehub/pull/43#discussion_r1427177526